engines/base/src/test/java/org/alfresco/transform/base/MtlsTestUtils.java

```diff
@@ -0,0 +1,69 @@
+package org.alfresco.transform.base;
+
+import org.apache.http.conn.ssl.SSLConnectionSocketFactory;
+import org.apache.http.conn.ssl.TrustAllStrategy;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.ssl.SSLContextBuilder;
+import org.springframework.http.client.ClientHttpRequestFactory;
+import org.springframework.http.client.HttpComponentsClientHttpRequestFactory;
+import org.springframework.web.client.RestTemplate;
+
+import javax.net.ssl.SSLContext;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.security.KeyManagementException;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.UnrecoverableKeyException;
+import java.security.cert.CertificateException;
+
+public class MtlsTestUtils {
+
+    public static boolean isMtlsEnabled()
+    {
+        return Boolean.parseBoolean(System.getProperty("mtls-enabled"));
+    }
+
+    public static CloseableHttpClient httpClientWithMtls() throws NoSuchAlgorithmException, KeyManagementException, UnrecoverableKeyException, KeyStoreException, IOException, CertificateException {
+
+        String keyStoreFile = System.getProperty("mtls-keystore-file");
+        String keyStoreType = System.getProperty("mtls-keystore-type");
+        char[] keyStorePassword = System.getProperty("mtls-keystore-password").toCharArray();
+        String trustStoreFile = System.getProperty("mtls-truststore-file");
+        String trustStoreType = System.getProperty("mtls-truststore-type");
+        char[] trustStorePassword = System.getProperty("mtls-truststore-password").toCharArray();
+
+        SSLContextBuilder sslContextBuilder = new SSLContextBuilder();
+        KeyStore keyStore = KeyStore.getInstance(keyStoreType);
+        try (InputStream keyStoreInputStream = new FileInputStream(keyStoreFile))
+        {
+            keyStore.load(keyStoreInputStream, keyStorePassword);
+            sslContextBuilder.loadKeyMaterial(keyStore, keyStorePassword);
+        }
+
+        KeyStore trustStore = KeyStore.getInstance(trustStoreType);
+        try (InputStream trustStoreInputStream = new FileInputStream(trustStoreFile))
+        {
+            trustStore.load(trustStoreInputStream, trustStorePassword);
+            sslContextBuilder.loadTrustMaterial(trustStore, TrustAllStrategy.INSTANCE);
+        }
+
+        SSLContext sslContext = sslContextBuilder.build();
+        SSLConnectionSocketFactory sslContextFactory = new SSLConnectionSocketFactory(sslContext);
+        return HttpClients.custom().setSSLSocketFactory(sslContextFactory).build();
+    }
+
+    public static RestTemplate restTemplateWithMtls()
+    {
+        ClientHttpRequestFactory requestFactory = null;
+        try {
+            requestFactory = new HttpComponentsClientHttpRequestFactory(httpClientWithMtls());
```

Comment:
Wait, so in tests the RestTemplate can be created using an HttpClient created with one version of initialising SSLContext and in config file it isn't that way? I want that config file to look similar to this, with 1 separate method responsible for creating an SslContext :D

Reply:
Those are different classes from different dependencies - SSLContext vs SslContext, HttpClient from apache vs HttpClient from netty. HttpClient configured for test classes here is different than the one configured for WebClient - and it has different usage here.

Reply:
I'd recommend moving the creation of the sslContext to a method called getApacheSslContext or something along the lines